node/node-sdk.md

```diff
@@ -7,193 +7,216 @@ The SDK steers developers to write their workflows and activities in TypeScript
+  * [Error Handling](#error-handling)
+  * [Cancellation](#cancellation)
+    * [Example](#example-1)
+    * [`CancellationScope.run()`](#cancellationscoperun)
+  * [Long Histories](#long-histories)
+* [Project Structure](#project-structure)
+  * [References](#references)
+
+<!-- vim-markdown-toc -->
 
 ## Activities
 
-### Interfaces
-To help users of your activities, it is recommended to define the activity interfaces in the `interfaces` sub-project.
-
-```typescript
-// interfaces/src/activities.ts
-
-export interface GreeterService {
-  greet(name: string): Promise<string>;
-}
-```
-
-### Implementations
-Activity implementations are just JS objects and functions, they run on the default NodeJS runtime, along with the Temporal SDK, and can import arbitrary npm packages.
+Activity implementations are just JS functions, they run in the default NodeJS runtime, along with the Temporal SDK's worker, and can import arbitrary npm packages.
 
 ### Example
-```typescript
-// activities/src/greeter.ts
-
-import { GreeterService } from '@interfaces/activities';
+```ts
+// activities/greeter.ts
 
-export const greeter: GreeterService = {
-  async greet(name: string) {
-    return `Hello, ${name}!`;
-  }
+export async function greet(name: string) {
+  return `Hello, ${name}!`;
 }
 ```
 
 ### `ActivityContext`
 
 Activities can get their `ActivityContext` by using `ActivityContext.current()`.
+
 This is implemented using [`AsyncLocalStorage`](https://nodejs.org/api/async_hooks.html#async_hooks_class_asynclocalstorage) which was introduced in `nodejs v13.10.0, v12.17.0`.
 (We can support earlier node versions by using the building blocks in [`async_hooks`](https://nodejs.org/api/async_hooks.html)).
 
-```typescript
-// activities/src/http.ts
+```ts
+// activities/http.ts
 
-import { HTTPService } from '@interfaces/activities'; // Definition omitted for brevity
 import { ActivityContext } from 'temporal-sdk';
 import axios from 'axios';
 
-export const http: HTTPService = {
-  async httpGet(url: string) {
-    const response = await axios.get(url, {
-      onDownloadProgress: (progressEvent) => {
-        const total = parseFloat(progressEvent.currentTarget.responseHeaders['Content-Length']);
-        const current = progressEvent.currentTarget.response.length;
-        const progress = Math.floor(current / total * 100);
-        ActivityContext.current().heartbeat(progress);
-      },
-    });
-    return response.data;
-  }
+export async httpGet(url: string): Promise<string> {
+  const response = await axios.get(url, {
+    onDownloadProgress: (progressEvent) => {
+      const total = parseFloat(progressEvent.currentTarget.responseHeaders['Content-Length']);
+      const current = progressEvent.currentTarget.response.length;
+      const progress = Math.floor(current / total * 100);
+      ActivityContext.current().heartbeat(progress);
+    },
+  });
+  return response.data;
 }
 ```
 
 ### Registration
```

Comment:
i had a new thought: should we have a concept of "local" vs "global" registration? i.e. if i register a workflow we may not want it to always be accessible by all other teams unless i want it to. somewhat of a security concern, but mostly i just believe that having good isolation between teams helps companies scale.

Reply:
The registration is always local in the sense that it only registered implementations with the local process. Currently, there is no concept of registration that the service understands. 

The cross-service communication is a big area that we should add later. It would include public/private interfaces as well as authorization primitives.

Reply:
OK. will mark as resolved for now knowing you're thinking about it